### PR TITLE
Support vstack function

### PR DESCRIPTION
**Pull request by fukatani**
This PR is for supporting `functions.vstack`. ( #1280 )

### CODE REVIEW

File: chainer/functions/array/vstack.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        ndim = in_types[0].ndim.eval()
+        for i in six.moves.range(1, in_types.size().eval()):
+            type_check.expect(
+                in_types[0].dtype == in_types[i].dtype,
+                in_types[0].ndim == in_types[i].ndim,
+            )
+            if ndim <= 1:
+                type_check.expect(in_types[0].shape == in_types[i].shape)
+                continue
+            for d in six.moves.range(0, ndim):
```
Comment (delta2323) [inline chainer/functions/array/vstack.py]: It is equivalent to `six.moves.range(ndim)`

Comment (delta2323) [inline chainer/functions/array/vstack.py]: `six.moves.range(1, ndim)` is simpler.

File: tests/chainer_tests/functions_tests/array_tests/test_vstack.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+@testing.parameterize(
+    {'a_shape': (2, 4, 5), 'b_shape': (3, 4, 5), 'valid': True},
+    {'a_shape': (3, 4, 6), 'b_shape': (3, 4, 5), 'valid': False},
+    {'a_shape': (3, 6, 5), 'b_shape': (3, 4, 5), 'valid': False},
+    {'a_shape': (3, 4), 'b_shape': (4, 4), 'valid': True},
+    {'a_shape': (3, 4), 'b_shape': (3, 3), 'valid': False},
+    {'a_shape': (3,), 'b_shape': (4,), 'valid': False},
+    {'a_shape': (3), 'b_shape': (3, 3), 'valid': False},
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_vstack.py]: Please add test case in which at least one of `a_shape` or `b_shape` is zero-dimensional.

**Comment by delta2323** [comment]
Some GPU tests fail. Could you check them if you have GPU environment? Be aware travis CI checks only CPU tests.

**Comment by fukatani** [comment]
Unfortunately, I don't have GPU environment.
If you could paste test log, I may be able to debug. (No self-confidence)

**Comment by delta2323** [comment]
I found that the failures originate from a bug in `cupy.vstack`. As I fix the problem in #1562. Could you wait until the PR is merged (or it is ok to merge the PR into yours).

**Comment by delta2323** [comment]
As #1562 is merged, could you merge or rebase the latest master branch?

**Comment by fukatani** [comment]
OK! Imerged it.

**Comment by fukatani** [comment]
I found same logic in `cupy.dstack`.
I afraid it not work as numpy.
(But no direct relationship with this PR)

**Comment by delta2323** [comment]
Thank you! I made a PR #1568

**Comment by delta2323** [comment]
LGTM 👍
